timeseries/src/autogluon/timeseries/models/local/abstract_local_model.py

```diff
@@ -58,6 +58,8 @@ def __init__(
     ):
         if hyperparameters is None:
             hyperparameters = {}
+        else:
+            hyperparameters = hyperparameters.copy()
```

Comment:
In-place modification of `hyperparameters` here led to a bug in `timeseries/tests/smoketests/test_all_models.py`: we popped the `"use_fallback_model"` key from `hyperparameters` (modifying the `DUMMY_MODEL_HPARAMS` dictionary in tests), so all models except the first one actually used the fallback model. 

Because of this, model failures for some models would go unnoticed by the tests.

Reply:
nice catch